[Bug] `loggingConfig.ts` Uses `pino-pretty` in Non-Production Environments — Will Crash if `pino-pretty` Is Not Installed

### Context
```ts
transport: process.env.NODE_ENV !== 'production'
  ? { target: 'pino-pretty', ... }
  : undefined,
```

`pino-pretty` is a dev dependency. If it is missing from `node_modules` (e.g., after `npm ci --omit=dev` in a staging environment that sets `NODE_ENV=staging`), the app will crash at startup with `Error: Cannot find module 'pino-pretty'`.

### Fix
1. Add `pino-pretty` to `dependencies` (not `devDependencies`) or guard the transport config:
```ts
const hasPinoPretty = (() => { try { require.resolve('pino-pretty'); return true; } catch { return false; } })();
transport: hasPinoPretty && process.env.NODE_ENV !== 'production' ? { target: 'pino-pretty', ... } : undefined,
```
2. Alternatively, only enable pretty-printing when `LOG_PRETTY=true` is explicitly set.

### Acceptance Criteria
- [ ] App starts without errors when `pino-pretty` is not installed
- [ ] Pretty logging is controlled by `LOG_PRETTY=true` env var
- [ ] `.env.example` documents `LOG_PRETTY`

--- a/backend/.env.example
+++ b/backend/.env.example
@@ -32,6 +32,7 @@ AWS_S3_BUCKET=your-s3-bucket-name
 # Logging Configuration
 NODE_ENV=development
 LOG_LEVEL=info
+LOG_PRETTY=true
 
 # KYC Configuration
 # Development-only: when true, non-corporate KYC submissions are auto-approved.

--- a/backend/src/auth/auth.controller.ts
+++ b/backend/src/auth/auth.controller.ts
@@ -63,4 +63,14 @@ export class AuthController {
   submitKyc(@Request() req: AuthRequest, @Body() dto: KycDto) {
     return this.authService.submitKyc(req.user.id, dto);
   }
+
+  @Post('logout')
+  @UseGuards(AuthGuard('jwt'))
+  @ApiBearerAuth('jwt')
+  @ApiOperation({ summary: 'Logout and invalidate the current JWT token' })
+  @ApiResponse({ status: 200, description: 'Logged out successfully' })
+  @ApiResponse({ status: 401, description: 'Unauthorized' })
+  logout(@Request() req: AuthRequest) {
+    return this.authService.logout(req.user.id);
+  }
 }

--- a/backend/src/auth/auth.service.ts
+++ b/backend/src/auth/auth.service.ts
@@ -233,4 +233,13 @@ export class AuthService {
     const saved = await this.userRepo.save(user);
     return { id: saved.id, role: saved.role };
   }
+
+  async logout(userId: string): Promise<{ message: string }> {
+    const user = await this.userRepo.findOne({ where: { id: userId } });
+    if (!user) throw new NotFoundException('User not found.');
+
+    user.tokenVersion = (user.tokenVersion ?? 0) + 1;
+    await this.userRepo.save(user);
+    return { message: 'Logged out successfully.' };
+  }
 }

--- a/backend/src/common/logging/logging.config.ts
+++ b/backend/src/common/logging/logging.config.ts
@@ -1,10 +1,24 @@
 import { Params } from 'nestjs-pino';
 
+/**
+ * Safely check if pino-pretty is available.
+ * This prevents crashes when pino-pretty is not installed
+ * (e.g., after npm ci --omit=dev in staging environments).
+ */
+function hasPinoPretty(): boolean {
+  try {
+    require.resolve('pino-pretty');
+    return true;
+  } catch {
+    return false;
+  }
+}
+
 export const loggingConfig: Params = {
   pinoHttp: {
     level: process.env.LOG_LEVEL || 'info',
     transport:
-      process.env.NODE_ENV !== 'production'
+      process.env.LOG_PRETTY === 'true' && hasPinoPretty()
         ? {
             target: 'pino-pretty',
             options: {

--- a/backend/src/queue/queue.processor.ts
+++ b/backend/src/queue/queue.processor.ts
@@ -5,6 +5,7 @@ import { Repository } from 'typeorm';
 import { PinoLogger } from 'nestjs-pino';
 import { StellarService } from '../stellar/stellar.service';
 import { TradeDealsService } from '../trade-deals/trade-deals.service';
+import { TradeDeal } from '../trade-deals/entities/trade-deal.entity';
 import { Investment } from '../investments/entities/investment.entity';
 import { User } from '../auth/entities/user.entity';
 import { NotificationsService } from '../notifications/notifications.service';
@@ -22,6 +23,8 @@ export class QueueProcessor {
   constructor(
     private readonly stellarService: StellarService,
     private readonly tradeDealsService: TradeDealsService,
+    @InjectRepository(TradeDeal)
+    private readonly tradeDealRepo: Repository<TradeDeal>,
     @InjectRepository(Investment)
     private readonly investmentRepo: Repository<Investment>,
     @InjectRepository(User)
@@ -61,12 +64,17 @@ export class QueueProcessor {
         data.tokenCount,
       );
 
-      // Update deal status to open and store stellar_asset_tx_id
-      await this.tradeDealsService.updateDealStatus(
-        data.dealId,
-        'open',
-        result.txId,
-      );
+      // Encrypt the issuer secret
+      const encryptedIssuerSecret =
+        this.stellarService.encryptSecret(result.issuerSecret);
+
+      // Update deal with issuer keys and status to open
+      await this.tradeDealRepo.update(data.dealId, {
+        status: 'open',
+        stellarAssetTxId: result.txId,
+        issuerPublicKey: result.issuerPublicKey,
+        issuerSecretKey: encryptedIssuerSecret,
+      });
 
       this.logger.info(
         { dealId: data.dealId, txId: result.txId },
@@ -79,7 +87,7 @@ export class QueueProcessor {
       );
 
       // On Stellar failure: mark deal status = 'failed'
-      await this.tradeDealsService.updateDealStatus(data.dealId, 'failed');
+      await this.tradeDealRepo.update(data.dealId, { status: 'failed' });
     }
 
     // Acknowledge the message

--- a/backend/src/trade-deals/trade-deals.controller.ts
+++ b/backend/src/trade-deals/trade-deals.controller.ts
@@ -8,6 +8,7 @@ import {
   UseGuards,
   Request,
   ForbiddenException,
+  HttpCode,
 } from '@nestjs/common';
 import {
   ApiTags,
@@ -60,7 +61,18 @@ export class TradeDealsController {
   }
 
   @Post(':id/publish')
+  @HttpCode(202)
   @UseGuards(AuthGuard('jwt'), KycGuard)
+  @ApiBearerAuth('jwt')
+  @ApiOperation({
+    summary: 'Publish a draft trade deal (async token issuance)',
+  })
+  @ApiResponse({ status: 202, description: 'Deal publish request accepted, token issuance in progress' })
+  @ApiResponse({ status: 400, description: 'Validation error' })
+  @ApiResponse({ status: 401, description: 'Unauthorized' })
+  @ApiResponse({ status: 403, description: 'Role or KYC requirement not met' })
+  @ApiResponse({ status: 404, description: 'Trade deal not found' })
+  @ApiResponse({ status: 422, description: 'Deal not in draft status or missing documents' })
   async publishDeal(
     @Param('id') id: string,
     @Request() req: AuthRequest,

--- a/backend/src/trade-deals/trade-deals.service.ts
+++ b/backend/src/trade-deals/trade-deals.service.ts
@@ -18,6 +18,7 @@ import {
   InvestmentStatus,
 } from '../investments/entities/investment.entity';
 import { StellarService } from '../stellar/stellar.service';
+import { QueueService } from '../queue/queue.service';
 import {
   normalizePagination,
   PaginatedResult,
@@ -57,6 +58,7 @@ export class TradeDealsService {
     @InjectRepository(Investment)
     private readonly investmentRepo: Repository<Investment>,
     private readonly stellarService: StellarService,
+    private readonly queueService: QueueService,
     private readonly logger: PinoLogger,
   ) {
     this.logger.setContext(TradeDealsService.name);
@@ -287,7 +289,7 @@ export class TradeDealsService {
     }
 
     try {
-      // Create escrow account
+      // Create escrow account synchronously (fast operation)
       this.logger.info({ dealId }, 'Creating escrow account for deal');
       const { publicKey: escrowPublicKey, secretKey: escrowSecretKey } =
         await this.stellarService.createEscrowAccount(dealId);
@@ -296,62 +298,43 @@ export class TradeDealsService {
       const encryptedEscrowSecret =
         this.stellarService.encryptSecret(escrowSecretKey);
 
-      // Issue trade token
-      this.logger.info(
-        { dealId, tokenSymbol: deal.tokenSymbol },
-        'Issuing trade token for deal',
-      );
-      const {
-        txId: stellarAssetTxId,
-        issuerPublicKey,
-        issuerSecret,
-      } = await this.stellarService.issueTradeToken(
-        deal.tokenSymbol,
-        escrowPublicKey,
-        escrowSecretKey,
-        deal.tokenCount,
-      );
-
-      // Encrypt the issuer secret
-      const encryptedIssuerSecret =
-        this.stellarService.encryptSecret(issuerSecret);
-
-      // Update deal with Stellar data
+      // Update deal with escrow data
       await this.tradeDealRepo.update(dealId, {
-        status: 'open',
         escrowPublicKey,
         escrowSecretKey: encryptedEscrowSecret,
-        issuerPublicKey,
-        issuerSecretKey: encryptedIssuerSecret,
-        stellarAssetTxId,
       });
 
       this.logger.info(
-        { dealId, txId: stellarAssetTxId, escrowPublicKey },
-        'Successfully published deal with Stellar integration',
+        { dealId, escrowPublicKey },
+        'Escrow account created, enqueuing token issuance',
       );
 
-      // Return updated deal
+      // Enqueue the token issuance job
+      await this.queueService.enqueueDealPublish({
+        dealId,
+        tokenSymbol: deal.tokenSymbol,
+        escrowPublicKey,
+        encryptedEscrowSecret,
+        tokenCount: deal.tokenCount,
+      });
+
+      // Return deal with escrow data (status still draft, will be updated by queue processor)
       return {
         ...deal,
-        status: 'open',
         escrowPublicKey,
         escrowSecretKey: encryptedEscrowSecret,
-        issuerPublicKey,
-        issuerSecretKey: encryptedIssuerSecret,
-        stellarAssetTxId,
       };
     } catch (error) {
       this.logger.error(
         { dealId, error: error.message },
-        'Failed to publish deal - Stellar operations failed',
+        'Failed to publish deal - escrow account creation failed',
       );
 
       // Deal remains in draft status on Stellar failure
       throw new UnprocessableEntityException({
         code: 'STELLAR_OPERATION_FAILED',
         message:
-          'Failed to create escrow account or issue trade token. Please try again.',
+          'Failed to create escrow account. Please try again.',
       });
     }
   }

--- a/frontend/src/app/dashboard/farmer/page.tsx
+++ b/frontend/src/app/dashboard/farmer/page.tsx
@@ -2,7 +2,7 @@
 
 import { useEffect, useState } from 'react';
 import { useRouter } from 'next/navigation';
-import { apiClient, Deal, User } from '@/lib/api';
+import { apiClient, Deal, User, MILESTONE_LABELS } from '@/lib/api';
 import ErrorBoundary from '@/components/ErrorBoundary';
 
 export default function FarmerDashboard() {
@@ -142,8 +142,14 @@ export default function FarmerDashboard() {
                   Welcome, {user?.name || user?.email}
                 </span>
                 <button
-                  onClick={() => {
-                    apiClient.clearAuth();
+                  onClick={async () => {
+                    try {
+                      await apiClient.logout();
+                    } catch (err) {
+                      console.error('Logout failed:', err);
+                      // Still clear auth and redirect even if logout fails
+                      apiClient.clearAuth();
+                    }
                     router.push('/login');
                   }}
                   className="bg-red-600 text-white px-4 py-2 rounded text-sm hover:bg-red-700 transition-colors"
@@ -232,7 +238,7 @@ export default function FarmerDashboard() {
                           {/* Latest Milestone */}
                           {latestMilestone && (() => {
                             const milestoneStatus = latestMilestone.milestone === 'importer' ? 'completed' : 'active';
-                            const milestoneLabel = latestMilestone.milestone.charAt(0).toUpperCase() + latestMilestone.milestone.slice(1);
+                            const milestoneLabel = MILESTONE_LABELS[latestMilestone.milestone];
                             return (
                               <div className="border-t pt-3">
                                 <div className="flex justify-between items-start">

--- a/frontend/src/app/dashboard/investor/page.tsx
+++ b/frontend/src/app/dashboard/investor/page.tsx
@@ -125,8 +125,14 @@ export default function InvestorDashboard() {
                   Welcome, {user?.name || user?.email}
                 </span>
                 <button
-                  onClick={() => {
-                    apiClient.clearAuth();
+                  onClick={async () => {
+                    try {
+                      await apiClient.logout();
+                    } catch (err) {
+                      console.error('Logout failed:', err);
+                      // Still clear auth and redirect even if logout fails
+                      apiClient.clearAuth();
+                    }
                     router.push('/login');
                   }}
                   className="bg-red-600 text-white px-4 py-2 rounded text-sm hover:bg-red-700 transition-colors"

--- a/frontend/src/app/dashboard/trader/page.tsx
+++ b/frontend/src/app/dashboard/trader/page.tsx
@@ -164,8 +164,14 @@ export default function TraderDashboard() {
                   Welcome, {user?.name || user?.email}
                 </span>
                 <button
-                  onClick={() => {
-                    apiClient.clearAuth();
+                  onClick={async () => {
+                    try {
+                      await apiClient.logout();
+                    } catch (err) {
+                      console.error('Logout failed:', err);
+                      // Still clear auth and redirect even if logout fails
+                      apiClient.clearAuth();
+                    }
                     router.push('/login');
                   }}
                   className="bg-red-600 text-white px-4 py-2 rounded text-sm hover:bg-red-700 transition-colors"

--- a/frontend/src/lib/api.ts
+++ b/frontend/src/lib/api.ts
@@ -25,9 +25,18 @@ export interface Document {
   created_at: string;
 }
 
+export type MilestoneType = "farm" | "warehouse" | "port" | "importer";
+
+export const MILESTONE_LABELS: Record<MilestoneType, string> = {
+  farm: "Farm",
+  warehouse: "Warehouse",
+  port: "Port",
+  importer: "Importer",
+};
+
 export interface Milestone {
   id: string;
-  milestone: "farm" | "warehouse" | "port" | "importer";
+  milestone: MilestoneType;
   notes: string | null;
   recorded_at: string;
   created_at: string;
@@ -259,6 +268,18 @@ export const apiClient = {
       method: "POST",
       body: JSON.stringify(data),
     });
+  },
+
+  // POST /auth/logout — invalidate the current JWT by incrementing tokenVersion
+  async logout(): Promise<{ message: string }> {
+    try {
+      return await apiFetch("/auth/logout", {
+        method: "POST",
+      });
+    } finally {
+      // Always clear local auth state, even if logout request fails
+      this.clearAuth();
+    }
   },
 };
 

